### ISSUE TITLE
Bug: just focus on rule being examined

### ISSUE THREAD

**Issue by nicoabie** [eslint/eslint#16554]
### Environment

Node version: 18.12.0
npm version: 8.19.2
Local ESLint version: 8.8.0
Global ESLint version: none
Operating System: macOS 13.0 (22A380)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
npx eslint --no-eslintrc --parser '@typescript-eslint/parser' --plugin '@typescript-eslint' --rule '@typescript-eslint/no-non-null-assertion: error' --ext .ts apps
```
</details>

### What did you expect to happen?

I expect eslint to report only on that rule. Show only errors for that

### What actually happened?

What eslint is doing is reporting on that rule and also telling me that has no definitions for rules I'm disabling in files.

For example using in a file that is being examinated
// eslint-disable-next-line import/no-import-module-exports

Shows
  1:1  error  Definition for rule 'import/no-import-module-exports' was not found  import/no-import-module-exports
 In the console
 
 I don't care about that

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

ESLINT is great!!!

**Comment by mdjermanovic** [comment]
Hi @nicoabie, thanks for the issue!

Can you also add the [--no-inline-config](https://eslint.org/docs/latest/user-guide/command-line-interface#--no-inline-config) flag to your command, and would that solve the problem?

**Comment by nzakas** [comment]
@nicoabie I'm glad that worked for you, but it did so by disabling all inline config comments, which I don't think is a good approach for everyone attempting to run a single rule.

@mdjermanovic do you think this is a bug? I know we don't search for rules when they are set to "off in a config file, and `eslint-disable-next-line` seems like maybe it should be treated the same way?

**Comment by btmills** [comment]
@nicoabie thanks for your enthusiasm for ESLint! We discussed this in our most recent team meeting and came to the conclusion that this is working as designed here. Most of the time when a rule in a disable comment doesn't exist, it's because of a typo or a version mismatch, for example, and we're optimizing for that case with this behavior.

Speaking for myself and not for the team now: there's an argument to be made that `/* eslint-disable non-existent-rule */` should instead be treated as an unused disable directive that would be caught by `--report-unused-disable-directives`. If you're interested in pursuing that behavior change rather than the workaround mdjermanovic suggested, I'd be willing to write up a new issue describing that idea so the team can discuss it.

**Comment by nicoabie** [comment]
@btmills I think just clarifying in the docs how to just run a single rule would be cool.
I believe it is a valid use case as a team to say let's fix this ESLINT rule for all files and you want to know in which files it is happening
